src/test/commands.vdb.test.ts

```diff
@@ -16,108 +16,63 @@
-			} catch (error) {
-				should.exist(error);
-			}
+		before( async () => {
+			workspacePath = vscode.workspace.workspaceFolders[0].uri.fsPath;
+			should.exist(workspacePath, `Unable to obtain the current workspace path.`);
+			workspacePath.should.contain('testFixture');
+			templateFolder = path.join(workspacePath, '../resources/');
 		});
 
-		it('should not generate a VDB file when passing invalid folder name', async () => {
-			try {
-				await createVDBCommand.handleVDBCreation(undefined, name, templateFolder);
-				fail('create command did not throw exception when passing invalid workspace path');
-			} catch (error) {
-				should.exist(error);
-			}
+		beforeEach( () => {
+			vdbFile = path.join(workspacePath, `${name}.yaml`);
+			cleanupVDB();
+			showInputBoxStub = sinon.stub(vscode.window, 'showInputBox');
+			commandSpy = sinon.spy(vscode.commands, 'executeCommand');
 		});
-	});
-
-	context('Delete VDB', () => {
 
-		beforeEach( async () => {
-			const success = await createVDBCommand.handleVDBCreation(workspacePath, name, templateFolder);
-			should.equal(true, success, 'Execution of the createVDBCommand returned false');
-			fs.existsSync(vdbFile).should.equal(true);
+		afterEach( () => {
+			showInputBoxStub.restore();
+			commandSpy.restore();
+			cleanupVDB();
 		});
 
-		it('should delete a VDB file when passing valid parameters', async () => {
-			const success = await deleteVDBCommand.handleVDBDeletion(name, vdbFile);
-			should.equal(true, success, `Execution of the deleteVDBCommand returned false for VDB ${name} with file ${vdbFile}`);
-			fs.existsSync(vdbFile).should.equal(false);
-			const dvConfig = utils.loadModelFromFile(vdbFile);
-			should.not.exist(dvConfig);
-		});
-
-		it('should not delete a VDB file when passing invalid file name', async () => {
-			try {
-				await deleteVDBCommand.handleVDBDeletion(name, `${vdbFile}.not.existing.yaml`);
-				fail('delete command did not throw exception when passing invalid file');
-			} catch (error) {
-				should.exist(error);
-			}
-		});
-
-		it('should not delete a VDB file when passing undefined as file', async () => {
-			try {
-				await createVDBCommand.handleVDBCreation(name, undefined);
-				fail('delete command did not throw exception when passing undefined instead of the file');
-			} catch (error) {
-				should.exist(error);
-			}
+		it('should generate a valid VDB file when handing over valid parameters and open the ddl in an editor', async () => {
+			showInputBoxStub.onFirstCall().returns(Promise.resolve(name));
+			await vscode.commands.executeCommand('datavirt.create.vdb');
+			fs.existsSync(vdbFile).should.equal(true, `The YAML file for VDB ${name} was not found.`);
+			const dvConfig = await utils.loadModelFromFile(vdbFile);
+			should.exist(dvConfig);
+			should.equal(dvConfig.metadata.name, name);
+			dvConfig.spec.build.source.ddl.should.contain(name);
+			commandSpy.should.have.been.calledWith('datavirt.edit.schema');
```

Comment:
checking that the editor is opened seems more robust than checking that this method has been called. If this method is failing we won't detect it.
This kind of check can be used for that:
https://github.com/camel-tooling/vscode-camelk/blob/fc725935285092c683a8e534b0a8343e6dfe6217/src/test/suite/commands/NewIntegrationFileCommand.test.ts#L92-L94

Reply:
Unfortunately that doesn't work as expected. I already did try that before I switched to spy the command. The activeEditor did not contain the expected value and I also check all available textdocuments without success. I will revisit that part when working on https://issues.redhat.com/browse/FUSETOOLS2-410

Reply:
The file name it displays to me is for instance `extension-output-#7` for the activeTextEditor.document. I am currently not sure why.

Reply:
ok, that is a timing issue. Even when using await it somehow doesn't wait until the editor is really opened. using the waitUntil fixed the isssue.